packages/tpetra/core/test/Map/Map_Bug2431.cpp

```diff
@@ -0,0 +1,347 @@
+
+    std::cout << message 
+              << ": After Tpetra::createOneToOne without TieBreak on Proc " 
+              << overlapMap->getComm()->getRank() 
+              << "; nGids = " << nonOverlapMap->getNodeNumElements() << "\n";
+
+    auto myidx_no =  nonOverlapMap->getMyGlobalIndices();
+    outStream << "IDS ON PROC " << comm->getRank() << ": ";
+    for (std::size_t idx=0; idx<myidx_no.size(); ++idx) {
+      outStream << myidx_no[idx] << ", ";
+    }
+    outStream << "\n";
+
+    // Discover how many unique GIDs we had initially
+    std::unordered_map<GO,int> uniqueGids;
+
+    for (auto i = arrP0.begin(); i != arrP0.end(); i++) {
+      if (uniqueGids.find(*i) != uniqueGids.end()) 
+        uniqueGids[*i]++;
+      else
+        uniqueGids[*i] = 1;
+    }
+    for (auto i = arrP1.begin(); i != arrP1.end(); i++) {
+      if (uniqueGids.find(*i) != uniqueGids.end()) 
+        uniqueGids[*i]++;
+      else
+        uniqueGids[*i] = 1;
+    }
+    for (auto i = arrP2.begin(); i != arrP2.end(); i++) {
+      if (uniqueGids.find(*i) != uniqueGids.end()) 
+        uniqueGids[*i]++;
+      else
+        uniqueGids[*i] = 1;
+    }
+    for (auto i = arrP3.begin(); i != arrP3.end(); i++) {
+      if (uniqueGids.find(*i) != uniqueGids.end()) 
+        uniqueGids[*i]++;
+      else
+        uniqueGids[*i] = 1;
+    }
+
+    GO ncopies = 0;
+    for (auto i = uniqueGids.begin(); i != uniqueGids.end(); i++)
+      if (i->second > 1) ncopies += (i->second - 1);
+
+    if (pid == 0) {
+      std::cout << "\n\n" << message
+                << ": Before Tpetra::createOneToOne, there are " 
+                << uniqueGids.size() << " ids, with " 
+                << ncopies << " copies.\n";
+      std::cout << message 
+                << ": After Tpetra::createOneToOne with TieBreak, there are " 
+                << nonOverlapMapTB->getGlobalNumElements() << " ids.\n";
+      std::cout << message 
+                << ": After Tpetra::createOneToOne without TieBreak, there are "
+                << nonOverlapMap->getGlobalNumElements() << " ids.\n";
+      std::cout << "\n\n";
+    }
+
+    // Check the results; number of unique GIDs should be the same in all Maps.
+    if (uniqueGids.size() != nonOverlapMapTB->getGlobalNumElements())
+      errorFlag = -1;
+    if (uniqueGids.size() != nonOverlapMap->getGlobalNumElements())
+      errorFlag = -1;
+  }
+  catch (std::logic_error err) {
+    outStream << err.what() << "\n";
+    errorFlag = -1000;
+  };
+
+  return errorFlag;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+
+int main(int argc, char *argv[]) {
+
+  typedef Tpetra::Map<>::local_ordinal_type LO;
+  typedef Tpetra::Map<>::global_ordinal_type GO;
+  typedef Tpetra::Map<>::node_type NO;
+
+  Teuchos::GlobalMPISession mpiSession(&argc, &argv);
+
+  Teuchos::RCP<const Teuchos::Comm<int>> comm = 
+           Tpetra::DefaultPlatform::getDefaultPlatform().getComm();
```

Comment:
Prefer `Tpetra::getDefaultComm()`.  It integrates better with `Tpetra::initialize`, and the various Platform classes are a relic of 2009 experimentation in heterogeneous computing (the one Platform class that could actually do that has long since been removed).